osgp/platform/osgp-core/src/test/java/org/opensmartgridplatform/core/application/services/EventNotificationMessageServiceTest.java

```diff
@@ -0,0 +1,71 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.opensmartgridplatform.core.application.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import org.joda.time.DateTime;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensmartgridplatform.core.domain.model.domain.DomainRequestService;
+import org.opensmartgridplatform.domain.core.entities.Event;
+import org.opensmartgridplatform.domain.core.exceptions.UnknownEntityException;
+import org.opensmartgridplatform.domain.core.valueobjects.EventType;
+import org.opensmartgridplatform.dto.valueobjects.EventNotificationDto;
+import org.opensmartgridplatform.dto.valueobjects.EventTypeDto;
+import org.opensmartgridplatform.shared.domain.services.CorrelationIdProviderTimestampService;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+
+@ExtendWith(MockitoExtension.class)
+public class EventNotificationMessageServiceTest {
+
+    @Mock
+    private CorrelationIdProviderTimestampService correlationIdProviderTimestampService;
+
+    @Mock
+    private DomainRequestService domainRequestService;
+
+    @Mock
+    private EventNotificationHelperService eventNotificationHelperService;
+
+    @InjectMocks
+    private EventNotificationMessageService eventNotificationMessageService;
+
+    @Test
+    void handleLightSensorReportsLightEventTest() throws UnknownEntityException {
+        final String deviceUid = "testUid";
+        final String deviceIdentification = "testIdentification";
+        final DateTime dateTime = DateTime.now();
+        final EventTypeDto eventTypeDto = EventTypeDto.LIGHT_SENSOR_REPORTS_LIGHT;
+        final String description = "Sensor reports light";
+        final Integer index = 0;
+        final EventNotificationDto eventNotificationDto = new EventNotificationDto(deviceUid, dateTime, eventTypeDto,
+                description, index);
+
+        final EventNotificationMessageService eventNotificationMessageServiceSpy = spy(
+                this.eventNotificationMessageService);
+        eventNotificationMessageServiceSpy.handleEvent(deviceIdentification, eventNotificationDto);
+
+        final ArgumentCaptor<String> argMessageType = ArgumentCaptor.forClass(String.class);
+        final ArgumentCaptor<String> argDeviceIdentification = ArgumentCaptor.forClass(String.class);
+        final ArgumentCaptor<Event> argLightMeasurementDeviceEvent = ArgumentCaptor.forClass(Event.class);
+
+        verify(eventNotificationMessageServiceSpy).sendRequestMessageToDomainPublicLighting(argMessageType.capture(),
```

Comment:
Test for the call "this.domainRequestService.send(...)", which is a call to another service instead of a private method. This would also simplify your test a bit.

Reply:
There are 2 paths that lead to this.domainRequestService.send. Only one path is valid,  therefore I think the current solution is a better test. Although a bit more complicated.